stream.go

```diff
@@ -28,9 +28,10 @@ type StreamState int
 
 // StreamState enums
 const (
-	StreamStateOpen    StreamState = iota // Stream object starts with StreamStateOpen
-	StreamStateClosing                    // Outgoing stream is being reset
-	StreamStateClosed                     // Stream has been closed
+	StreamStateOpen          StreamState = iota // Stream object starts with StreamStateOpen
+	StreamStateClosing                          // Outgoing stream is being reset
```

Comment:
As [discussed earlier](https://github.com/pion/sctp/issues/239#issuecomment-1284985062), stream.Close() should close both inbound and outbound. (the state will immediately goes to StreamStateClosed)

When s.Close() is called, we should set [s.readErr to io.EOF](https://github.com/pion/sctp/blob/d0b7cf3963e99337bdf09e11989cf8b0916e1179/stream.go#L282) then [call s.readNotifier.Broadcast()](https://github.com/pion/sctp/blob/d0b7cf3963e99337bdf09e11989cf8b0916e1179/stream.go#L286) as we used to do.

Reply:
@enobufs  Updated.

Reply:
@jerry-tao Are you interested in fixing the failing tests?  If not, I can help you this weekend.

Reply:
I'm working on the failing test, I will commit the code later.
But in our browser test, it failed again and stuck at closing, working on it now.